src/mongoose_server_api.erl

```diff
@@ -0,0 +1,156 @@
+-module(mongoose_server_api).
+
+-export([get_loglevel/0, status/0, get_cookie/0, join_cluster/1, leave_cluster/0,
+         remove_from_cluster/1, stop/0, restart/0, remove_node/1, set_loglevel/1,
+         graphql_get_loglevel/0]).
+
+-ignore_xref([get_loglevel/0]).
+
+-spec get_loglevel() -> {ok, iodata()}.
+get_loglevel() ->
+    Level = mongoose_logs:get_global_loglevel(),
+    Number = mongoose_logs:loglevel_keyword_to_number(Level),
+    String = io_lib:format("global loglevel is ~p, which means '~p'", [Number, Level]),
+    {ok, String}.
+
+-spec graphql_get_loglevel() -> {ok, mongoose_logs:atom_log_level()}.
+graphql_get_loglevel() ->
+    {ok, mongoose_logs:get_global_loglevel()}.
+
+
+-spec set_loglevel(mongoose_logs:atom_log_level()) ->
+    {ok, iodata()} | {invalid_level, iodata()}.
+set_loglevel(Level) ->
+    case mongoose_logs:set_global_loglevel(Level) of
+        ok ->
+            {ok, "Log level successfully set"};
+        {error, _} ->
+            {invalid_level, io_lib:format("Log level ~p does not exist", [Level])}
+    end.
+
+-spec status() -> {'mongooseim_not_running', io_lib:chars()} | {'ok', io_lib:chars()}.
+status() ->
+    {InternalStatus, ProvidedStatus} = init:get_status(),
+    String1 = io_lib:format("The node ~p is ~p. Status: ~p",
+                            [node(), InternalStatus, ProvidedStatus]),
+    case lists:keysearch(mongooseim, 1, application:which_applications()) of
+        false ->
+            {mongooseim_not_running, String1 ++ "mongooseim is not running in that node."};
+        {value, {_, _, Version}} ->
+            {ok, String1 ++ io_lib:format("mongooseim ~s is running in that node", [Version])}
+    end.
+
+-spec get_cookie() -> string().
+get_cookie() ->
+    atom_to_list(erlang:get_cookie()).
+
+-spec join_cluster(string()) -> {ok, string()} | {pang, string()} | {already_joined, string()} |
```

Comment:
Please check the types for all functions, bacause I see that in some cases you are returning `iolist()` instead of `string()`. Apparently dialyzer does not complain, because `string` is a subtype of `iolist`, and a success path exists, but the types are wrong.